plugins/kafka/README.md

```diff
@@ -0,0 +1,233 @@
+
+    partitionCount := c.GetTopicPartitionCount();
+
+`GetTopicPartitionCount()` returns zero if the topic has not been created or there are has been an error.
+
+##Complete ECL Examples
+
+The following code will publish 100K messages to a topic named 'MyTestTopic' on an Apache Kafka broker located at address 10.211.55.13.  If you are running a single-node HPCC cluster and have installed Kafka on the same node, you can use 'localhost' instead (or omit the parameter, as it defaults to 'localhost').
+
+###Publishing
+
+    IMPORT kafka;
+
+    MyDataLayout := RECORD
+        STRING  message;
+    END;
+
+    ds := DATASET
+        (
+            100000,
+            TRANSFORM
+                (
+                    MyDataLayout,
+                    SELF.message := 'Test message ' + (STRING)COUNTER
+                ),
+            DISTRIBUTED
+        );
+
+    p := kafka.Publisher('MyTestTopic', brokers := '10.211.55.13');
+
+    APPLY(ds, ORDERED(p.PublishMessage(message)));
+
+###Consuming
+
+This code will read the messages written by the publishing example, above.  It will also show the number of partitions in the topic and the offsets of the last-read messages.
+
+    IMPORT kafka;
+
+    c := kafka.Consumer('MyTestTopic', brokers := '10.211.55.13');
+
+    ds := c.GetMessages(200000);
+    offsets := c.LastMessageOffsets(ds);
+    partitionCount := c.GetTopicPartitionCount();
+
+    OUTPUT(ds, NAMED('MessageSample'));
+    OUTPUT(COUNT(ds), NAMED('MessageCount'));
+    OUTPUT(offsets, NAMED('LastMessageOffsets'));
+    OUTPUT(partitionCount, NAMED('PartitionCount'));
+
+###Resetting Offsets
+
+Resetting offsets is useful when you have a topic already published with messages and you need to reread its messages from the very beginning.
+
+    IMPORT kafka;
+
+    c := kafka.Consumer('MyTestTopic', brokers := '10.211.55.13');
+
+    c.ResetMessageOffsets();
+
+##Behaviour and Implementation Details
+
+###Partitioning within Apache Kafka Topics
+
+Topic partitioning is covered in Apache Kafka's [introduction](https://kafka.apache.org/documentation.html#introduction).  There is a performance relationship between the number of partitions in a topic and the size of the HPCC cluster when consuming messages.  Ideally, the number of partitions will exactly equal the number of HPCC nodes consuming messages.  For Thor, this means the total number of slaves rather than the number of nodes, as that can be different in a multi-slave setup.  For Roxie, the number is always one.  If there are fewer partitions than nodes (slaves) then not all of your cluster will be utilized when consuming messages; if there are more partitions than nodes (slaves) then some nodes will be performing extra work, consuming from multiple partitions.  In either mismatch case, you may want to consider using the ECL DISTRIBUTE() function to redistribute your data before processing.
```

Comment:
Not a biggie but I think (if I am reading this correctly) that you have some very long lines in the readme file - these will be rendered fine by GitHub or other markdown viewers but may make it harder to view the file in other programs.